kiva/fonttools/_scanner.py

```diff
@@ -0,0 +1,538 @@
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+"""
+####### NOTE #######
+This is based heavily on matplotlib's font_manager.py rev 8713,
+but has been modified to not use other matplotlib modules
+####################
+
+Authors   : John Hunter <jdhunter@ace.bsd.uchicago.edu>
+            Paul Barrett <Barrett@STScI.Edu>
+            Michael Droettboom <mdroe@STScI.edu>
+Copyright : John Hunter (2004,2005), Paul Barrett (2004,2005)
+License   : matplotlib license (PSF compatible)
+            The font directory code is from ttfquery,
+            see license/LICENSE_TTFQUERY.
+"""
+import glob
+import logging
+import os
+import subprocess
+import sys
+
+from fontTools.ttLib import TTCollection, TTFont, TTLibError
+
+from kiva.fonttools import afm
+from kiva.fonttools._constants import weight_dict
+from kiva.fonttools._util import get_ttf_prop_dict, weight_as_number
+
+logger = logging.getLogger(__name__)
+# Error message when fonts fail to load
+_FONT_ENTRY_ERR_MSG = "Could not convert font to FontEntry for file %s"
+
+# OS Font paths
+MSFolders = r"Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
+MSFontDirectories = [
+    r"SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts",
+    r"SOFTWARE\Microsoft\Windows\CurrentVersion\Fonts",
+]
+OSXFontDirectories = [
+    "/Library/Fonts/",
+    "/Network/Library/Fonts/",
+    "/System/Library/Fonts/",
+]
+X11FontDirectories = [
+    # an old standard installation point
+    "/usr/X11R6/lib/X11/fonts/TTF/",
+    # here is the new standard location for fonts
+    "/usr/share/fonts/",
+    # documented as a good place to install new fonts
+    "/usr/local/share/fonts/",
+    # common application, not really useful
+    "/usr/lib/openoffice/share/fonts/truetype/",
+]
+
+home = os.environ.get("HOME")
+if home is not None:
+    # user fonts on OSX
+    path = os.path.join(home, "Library", "Fonts")
+    OSXFontDirectories.append(path)
+    path = os.path.join(home, ".fonts")
+    X11FontDirectories.append(path)
+
+
+class FontEntry(object):
+    """ A class for storing Font properties. It is used when populating
+    the font lookup dictionary.
+    """
+    def __init__(self, fname="", name="", style="normal", variant="normal",
+                 weight="normal", stretch="normal", size="medium"):
+        self.fname = fname
+        self.name = name
+        self.style = style
+        self.variant = variant
+        self.weight = weight
+        self.stretch = stretch
+        try:
+            self.size = str(float(size))
+        except ValueError:
+            self.size = size
+
+    def __repr__(self):
+        fname = os.path.basename(self.fname)
+        return (
+            f"<FontEntry '{self.name}' ({fname}) {self.style} {self.variant} "
+            f"{self.weight} {self.stretch}>"
+        )
+
+
+def create_font_list(fontfiles, fontext="ttf"):
```

Comment:
Is it necessary to specify that this function only handles `fontext` - `ttf` and `afm`? ~Or does it also implicitly handle `ttc` font files?~ looking at more code in this module, looks like `ttf` and `ttc` are handled by the same function